rust/lib.rs

```diff
@@ -0,0 +1,421 @@
+use serde_json::Value;
+use web3::{api::Eth, contract::Contract, transports::Http};
+
+const TESTNET_ADDRESS: &str = include_str!("../publish/testnet.json");
+const MOONBEAM_ADDRESS: &str = include_str!("../publish/moonbase.json");
+
+#[derive(Hash, Eq, PartialEq, Copy, Clone, Debug)]
+pub enum Network {
+    Testnet,
+    Moonbeam,
+    Mainnet,
+}
+
+impl Network {
+    fn address(&self) -> &str {
+        match self {
+            Network::Testnet => TESTNET_ADDRESS,
+            Network::Moonbeam => MOONBEAM_ADDRESS,
+            Network::Mainnet => TESTNET_ADDRESS,
+        }
+    }
+}
+
+pub fn settings(eth: Eth<Http>, network: Network) -> Result<Contract<Http>, ()> {
```

Comment:
[suggestion] Looks these contract initialisation can share the same logic with extra `contract_name` parameter.

```
let path =  format!("../artifacts/contracts/{contract_name}.sol/{contract_name}.json");
address[contract_name]["address"]
```

Reply:
Because `include_str` is hardcode, can not use compile-time or running time variables.